Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -97,7 +97,8 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
             guard connectionStatus != oldValue else {
                 return
             }
-            if case .connected = connectionStatus {
+            if case .connected = connectionStatus,
+               self.settings.notifyStatusChanges {
```

Comment:
We're now checking if notifications are enabled before presenting them.

Reply:
Does this make `NetworkProtectionNotificationsPresenterTogglableDecorator` within the iOS implementation redundant?

Reply:
I was surprised to see iOS not using `TunnelSettings` for a newly introduced setting to be honest, as I had shared this class a while back as a mechanism to add settings that can be supported across platforms.

This is in part why I would like some of these projects to be run across both platforms simultaneously.

Reply:
It doesn't make the decorator unnecessary, though.  I can switch to using the decorator for macOS too.

I'll push that.